### PR TITLE
Deprecate release infrastructure and doc - moved to kubernetes/release

### PR DESCRIPTION
**Pull request by david-mcmahon**
Part 2 of https://github.com/kubernetes/release/pull/1
This PR finalizes the split between the main kubernetes repo and the release tooling now under kubernetes/release.

ref #16529

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/24629)

### CODE REVIEW

File: docs/devel/releasing.md [hunk 1]
```diff
@@ -1,307 +0,0 @@
...
-1. check "This is a pre-release" if it's an alpha or beta release;
-1. fill in the release title from the draft;
-1. re-run the appropriate release notes tool(s) to pick up any changes people
-   have made;
-1. find the appropriate `kubernetes.tar.gz` in [GCS bucket](https://console.developers.google.com/storage/browser/kubernetes-release/release/),
-   download it, double check the hash (compare to what you had in the release
-   notes draft), and attach it to the release; and
-1. publish!
-
-### Manual tasks for new release series
```
Comment (ihmccreery) [inline docs/devel/releasing.md]: AFAIK, you haven't replaced any of this stuff in your new tooling, so it should probably stay/get moved over.

**Comment by ihmccreery** [comment]
LGTM; I assume you don't want to merge it yet, though.

**Comment by david-mcmahon** [comment]
Damn bot assigning!  I just wanted to get the PR up into the mix, but it's definitely not ready for merging yet.  It's not even completely done yet.  Sorry to bother..

File: docs/devel/releasing.md (hunk 1 above)
Comment (david-mcmahon) [inline docs/devel/releasing.md]: 1 is handled in the tooling and 2 is on the test team for now.  When there's an interface we can work with to have the release tooling trigger a new set of jobs, we will add that to the process.

**Comment by ihmccreery** [comment]
> Damn bot assigning! I just wanted to get the PR up into the mix, but it's definitely not ready for merging yet. It's not even completely done yet. Sorry to bother..

No problem.  In the future, you can assign a PR to yourself until it's ready for review.  Or just label it "[WIP]".  Or both.

**Comment by david-mcmahon** [comment]
Oh, will WIP preclude the autoassigning?  Good to know.

**Comment by ihmccreery** [comment]
> Oh, will WIP preclude the autoassigning? Good to know.

No, but it will make it clear it's a WIP and I shouldn't review it yet.  Best to assign to yourself to keep the assigner-bot away.

**Comment by spxtr** [comment]
Is this still WIP? I'd love it to go in soon.

**Comment by david-mcmahon** [comment]
@spxtr I'm eager myself. I do want to get those entry points the CI is using out of there too and working from the release repo first.

**Comment by fejta** [comment]
Please add back assignees when it is ready for more review

**Comment by david-mcmahon** [comment]
This is ready.  PTAL.

**Comment by ixdy** [comment]
So basically this leaves `build/release.sh` and everything it uses, but removes the rest?

**Comment by david-mcmahon** [comment]
@ixdy yes, the remaining "release" bits are the parts of the build that construct the artifacts that make up the release.

File: build/common.sh [hunk 2]
```diff
@@ -1061,537 +1031,8 @@ function kube::release::create_tarball() {
...
-function kube::release::gcloud_account_is_active() {
-  local -r account="${1-}"
-  if [[ "$(gcloud config list --format='value(core.account)')" == "${account}" ]]; then
-    return 0
-  else
-    return 1
-  fi
-}
+###############################################################################
+# The ::release:: namespace functions have been moved to
```
Comment (ixdy) [inline build/common.sh]: they didn't all move (hence my comment) - maybe update this comment to note that the actual release functions have moved, but the packaging ones (related to build) remain.

Comment (ixdy) [inline build/common.sh]: (not sure if we intend to rename the ::release:: functions here to something like ::package::?)

**Comment by ixdy** [comment]
LGTM besides comment suggestion

**Pull request merged**